src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -308,6 +308,11 @@ private void TryAcquireWriteLock()
                 _writeLockFile.SetLength(1);
                 _writeLockFile.Lock(0, 1);
             }
+            catch (PlatformNotSupportedException)
```

Comment:
I'm wondering if we want to add `when (PlatformDetails.IsMacOs)` here?

Reply:
No, any `PlatformNotSupported` would work here. 
I'm thinking about BSD in the future.